tests/containers/podman_integration.pm

```diff
@@ -12,19 +12,46 @@ use testapi;
 use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
 use containers::utils qw(get_podman_version);
 use utils qw(zypper_call script_retry);
-use version_utils qw(get_os_release);
+use version_utils qw(get_os_release is_transactional is_sle is_sle_micro is_tumbleweed);
+use transactional qw(trup_call check_reboot_changes);
+use registration qw(add_suseconnect_product get_addon_fullname);
 use containers::common;
 
 sub run {
     select_serial_terminal;
     my ($running_version, $sp, $host_distri) = get_os_release;
     install_podman_when_needed($host_distri);
 
+    if (is_sle_micro) {
+        my $sle_version = "";
+        if (is_sle_micro('<5.3')) {
+            $sle_version = "15.3";
+        } elsif (is_sle_micro('<5.5')) {
+            $sle_version = "15.4";
+        } elsif (is_sle_micro('<6.0')) {
+            $sle_version = "15.5";
+        }
+        trup_call "register -p PackageHub/$sle_version/" . get_required_var('ARCH');
+        check_reboot_changes;
+        zypper_call "--gpg-auto-import-keys ref";
+    } elsif (is_sle) {
+        add_suseconnect_product(get_addon_fullname('phub'));
+    }
+
     # Install tests dependencies
-    my @pkgs = qw(apache2-utils bats buildah criu go gpg2 jq make netcat-openbsd openssl podman-remote python3-PyYAML skopeo socat sudo systemd-container);
-    zypper_call "in @pkgs";
+    my @pkgs = qw(bats jq make netcat-openbsd openssl python3-PyYAML socat sudo systemd-container);
+    push @pkgs, qw(apache2-utils buildah criu go gpg2 podman-remote skopeo) unless is_sle_micro;
+    if (is_transactional) {
+        trup_call "pkg install -y @pkgs";
+        check_reboot_changes;
+    } else {
+        zypper_call "in @pkgs";
+    }
+
+    # TODO: Create an user, add to tty group, subuid, etc on SLE Micro
```

Comment:
That's just

```
        assert_script_run "useradd -m $username -c '$realname'";
        assert_script_run "echo $username:$password | chpasswd";
```

i.e. simpler than skipping the rootless tests now

Reply:
It's a lot more messier than that:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/containers/rootless_podman.pm#L95

Reply:
Not sure how that's related - that workaround for the `script_run` design flaw is needed in any case also on non-SLEM?

Reply:
> Not sure how that's related - that workaround for the `script_run` design flaw is needed in any case also on non-SLEM?

Hopefully not.

Reply:
Fixed: https://openqa.suse.de/tests/13257992#